Agoda.Frameworks.DB/DbRepository.cs

```diff
@@ -303,14 +303,16 @@ protected virtual ShouldRetryPredicate ShouldRetry(int maxAttemptCount)
             string storedProc,
             int timeoutSecs,
             int maxAttemptCount,
-            CancellationToken token,
+            int taskCancellationTimeOutInMilliSecs,
             IDbDataParameter[] parameters,
             Func<SqlDataReader, Task<T>> callback)
         {
             return _dbResources.ChooseDb(database).ExecuteAsync(async (connectionStr, _) =>
             {
                 var stopwatch = Stopwatch.StartNew();
                 Exception error = null;
+                var cancellationTokenSource = new CancellationTokenSource();
+                cancellationTokenSource.CancelAfter(taskCancellationTimeOutInMilliSecs);
```

Comment:
What is the operation that you want to include in the timer? Only the query execution, or opening the connection as well? If the first then should we initialize this closer to where it's consumed? Or if you want to timeout even if opening the connection takes a long time, then should we pass the token to connection.OpenAsync() as well?